era_5g_interface/interface_helpers.py

```diff
@@ -100,37 +112,87 @@ def __init__(self) -> None:
         self.session.mount(MIDDLEWARE_ADDRESS, self.adapter)
         self.connection_error = False
 
-    def _send_middleware_heart_beat_request(self, headers: Dict, json: Dict) -> None:
+    def _send_heart_beat_request(self, headers: Dict, json: Dict, repeat_on_error: bool = False) -> None:
         """Send heart beat request to Middleware.
 
         Args:
             headers (Dict): Request headers.
             json (Dict): Request JSON.
+            repeat_on_error (bool): If set, repeat requests further. Defaults to False.
         """
 
-        if not self.connection_error:
+        if not self.connection_error or repeat_on_error:
             logger.debug(f"Sending heart beat to middleware: {json}")
             try:
                 response = self.session.post(MIDDLEWARE_ADDRESS, headers=headers, json=json, timeout=(0.2, 0.2))
                 if response.ok:
                     logger.debug(f"Middleware heart beat response: {response.text}")
                 else:
-                    logger.warning(f"Middleware heart beat response: {response}")
+                    logger.warning(
+                        f"Middleware heart beat response: {response}, middleware address: {MIDDLEWARE_ADDRESS}"
+                    )
                     self.connection_error = True
             except requests.RequestException as ex:
                 logger.warning(f"Failed to connect to the middleware address: {MIDDLEWARE_ADDRESS}, {repr(ex)}")
                 self.connection_error = True
 
+    def send_robot_heart_beat(
```

Comment:
Should this be part of era_5g_interface? I would assume it's just part of the heartbeat module...

Reply:
It may well be. The only thing that is done here is the creation of a dictionary from variables. For now it's like this because I don't know in what format the data will be sent and what it will have to do with "application heartbeat".

Reply:
ok